sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/DoNotCheckZeroSizeCollectionBase.cs

```diff
@@ -0,0 +1,147 @@
+/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class DoNotCheckZeroSizeCollectionBase<TLanguageKindEnum,
+            TBinaryExpressionSyntax,
+            TExpressionSyntax> : SonarDiagnosticAnalyzer
+        where TLanguageKindEnum : struct
+        where TBinaryExpressionSyntax : SyntaxNode
+        where TExpressionSyntax : SyntaxNode
+    {
+        protected const string DiagnosticId = "S3981";
+        protected const string MessageFormat =
+            "The {0} of '{1}' is always '>=0', so fix this test to get the real expected behavior.";
+
+        protected abstract DiagnosticDescriptor Rule { get; }
```

Comment:
Not sure whether this is a bug of GitHub but the colorization starting from this line looks like string instead of proper properties and so on. Could you confirm this is just a bug of GitHub?

Reply:
It's a github bug. Otherwise, even if it compiled, tests would fail :)